synapse/app/complement_fork_starter.py

```diff
@@ -102,21 +107,29 @@ def _worker_entrypoint(
 
     from twisted.internet.epollreactor import EPollReactor
 
-    proxy_reactor.___install(EPollReactor())
+    proxy_reactor._install_real_reactor(EPollReactor())
     func()
 
 
 def main() -> None:
     """
     Entrypoint for the forking launcher.
     """
-
-    # First argument is the path to the database config
-    db_config_path = sys.argv[1]
+    parser = argparse.ArgumentParser()
+    parser.add_argument("db_config", help="Path to database config file")
+    parser.add_argument(
+        "args",
+        nargs="...",
+        help="Argument groups separated by `--`. "
+        "The first argument of each group is a Synapse app name. "
+        "Subsequent arguments are passed through.",
+    )
+    ns = parser.parse_args()
+    db_config_path = ns.db_config
+    args = ns.args
 
     # Split up the subsequent arguments into each workers' arguments;
     # `--` is our delimiter of choice.
-    args = sys.argv[2:]
     args_by_worker: List[List[str]] = [
         list(args)
         for cond, args in itertools.groupby(args, lambda ele: ele != "--")
```

Comment:
can you avoid using `args` to refer to two different things? probably easier just to:

```suggestion
        for cond, args in itertools.groupby(ns.args, lambda ele: ele != "--")
```